src/v/cloud_storage/manifest.h

```diff
@@ -30,6 +30,51 @@
 
 namespace cloud_storage {
 
+/// Information contained inside the partition manifest path
+struct manifest_path_components {
+    std::filesystem::path _origin;
+    model::ns _ns;
+    model::topic _topic;
+    model::partition_id _part;
```

Comment:
Nit: would be nice to zero initialize the integer fields with a `{0}`, even if the code for parsing is correct today it makes bugs in future easier to handle by having a known empty value.

Reply:
@jcsp we sort of this: numeric named_types always initialize to `std:numeric_limits<wrapped_type>::min()`